libraries/ESP8266mDNS/src/LEAmDNS.cpp

```diff
@@ -1088,8 +1088,8 @@ bool MDNSResponder::setServiceProbeResultCallback(const MDNSResponder::hMDNSServ
 
 bool MDNSResponder::setServiceProbeResultCallback(const MDNSResponder::hMDNSService p_hService,
                                                   MDNSResponder::MDNSServiceProbeFn1 p_fnCallback) {
-	using namespace std::placeholders;
-	return setServiceProbeResultCallback(p_hService, std::bind(p_fnCallback, std::ref(*this), _1, _2, _3));
+    using namespace std::placeholders;
+    return setServiceProbeResultCallback(p_hService, [obj=std::ref(*this), p_fnCallback](const char* _1, const hMDNSService _2, bool _3) { p_fnCallback(obj, _1, _2, _3); });

```

Comment:
What is obj? Prefer a more descriptive name.

Reply:
Changing according to signature
```
typedef std::function<void(MDNSResponder& resp,
								   const char* p_pcServiceName,
								   const hMDNSService p_hMDNSService,
								   bool p_bProbeResult)> MDNSServiceProbeFn1;
```